### PR TITLE
added homework 14 for lesson 16 group 2

### PR DESCRIPTION
**Pull request by bkisel**

### CODE REVIEW

File: page-object/pages/SearchPage.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import { BasePage } from "../BasePage";
+
+
+export class SearchPage extends BasePage {
+searchPageLocator: any;
+  readonly searchField: Locator;
+  readonly filterByDateBtn: Locator;
+  readonly filterByViewedBtn: Locator;
+  readonly filterByCreatorBtn: Locator;
+  readonly filterByDatasetSizeBtn: Locator;
```
Comment (lifeisbeautifful) [inline page-object/pages/SearchPage.ts]: If I got it right, there are 3 dataset buttons with different sizes, so I suggest to change the name to smth like DatasetSmallSizeBtn

File: page-object/pages/SearchPage.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  readonly filterByFileTypeBtn: Locator;
```
Comment (lifeisbeautifful) [inline page-object/pages/SearchPage.ts]: The same: suggest to specify the name of filter type(csv)

File: page-object/pages/SearchPage.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  readonly filterByLicenseBtn: Locator;
```
Comment (lifeisbeautifful) [inline page-object/pages/SearchPage.ts]: The same, please specify the license type in button name

File: tests/kaggleSearch.spec.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    await kagglePage.filterByDatasetSize();
+    await expect(kagglePage.filterByDatasetSizeBtn).toBeVisible();
+
+    // Set Dataset filter
+    await kagglePage.filterByFileType();
+    await expect(kagglePage.filterByFileTypeBtn).toBeVisible();
+
+    // // Set License filter
+    // await kagglePage.filterByLicense();
+    // await expect(kagglePage.filterByLicenseBtn).toBeVisible();
```
Comment (lifeisbeautifful) [inline tests/kaggleSearch.spec.ts]: Why this test is committed?

File: tests/kaggleSearch.spec.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    // Перевірити наявність результатів
+    await expect(kagglePage.searchResultItem).toHaveCount(1);
+  });
+  test("should display 'No results found' page when no results found", async ({ page }) => {
+    const kagglePage = new SearchPage(page);
+
+    // Search (not existing value)
+    await kagglePage.searchFor('Sed ut perspiciatis.');
```
Comment (lifeisbeautifful) [inline tests/kaggleSearch.spec.ts]: Looks like the test data is outdated, search result is present and test fail

**Review by lifeisbeautifful — APPROVED** [review]

**Review by lifeisbeautifful — COMMENTED** [review]
Approved with suggestions

**Review by lifeisbeautifful — CHANGES_REQUESTED** [review]
I left comments and test review options

File: tests/kaggleSearch.spec.ts (hunk 5 above)
Comment (holllyrash) [reply tests/kaggleSearch.spec.ts]: thanks, fixed!

File: tests/kaggleSearch.spec.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  test.only("should display 'No results found' page when no results found", async ({ page }) => {
+    const kagglePage = new SearchPage(page);
+
+    // Search (not existing value)
+    await kagglePage.searchFor('якийсь текст українською');
+    await expect(kagglePage.searchField).toHaveValue('якийсь текст українською');
+
+
+    //Check for content 
+    await expect(kagglePage.noResultsFoundContentTitle).toContainText('No results found for ');
```
Comment (alexredko0004) [inline tests/kaggleSearch.spec.ts:42]: This test fails all the time for me on 'toContainText' check. It receives an empty string all the time

**Review by alexredko0004 — APPROVED** [review]
Done with comment about failing test

**Comment by Studigrad** [comment]
І тут конфліктики)

File: page-object/pages/SearchPage.ts (hunk 1 above)
Comment (bkisel) [reply page-object/pages/SearchPage.ts]: Done

File: page-object/pages/SearchPage.ts (hunk 2 above)
Comment (bkisel) [reply page-object/pages/SearchPage.ts]: Done

File: page-object/pages/SearchPage.ts (hunk 3 above)
Comment (bkisel) [reply page-object/pages/SearchPage.ts]: Done

File: tests/kaggleSearch.spec.ts (hunk 4 above)
Comment (bkisel) [reply tests/kaggleSearch.spec.ts]: Deleted

**Comment by bkisel** [comment]
@lifeisbeautifful please review and approve to unblock merging

**Review by Studigrad — APPROVED** [review]

**Review by LitvinenkoAnn — APPROVED** [review]

**Review by lifeisbeautifful — APPROVED** [review]

**Pull request merged**